staging/src/k8s.io/csi-translation-lib/plugins/azure_file.go

```diff
@@ -163,31 +163,48 @@ func (t *azureFileCSITranslator) TranslateCSIPVToInTree(pv *v1.PersistentVolume)
 		ReadOnly: csiSource.ReadOnly,
 	}
 
+	for k, v := range csiSource.VolumeAttributes {
+		switch strings.ToLower(k) {
+		case shareNameField:
+			azureSource.ShareName = v
+		case secretNameField:
+			azureSource.SecretName = v
+		case secretNamespaceField:
+			ns := v
+			azureSource.SecretNamespace = &ns
+		}
+	}
+
 	resourceGroup := ""
 	if csiSource.NodeStageSecretRef != nil && csiSource.NodeStageSecretRef.Name != "" {
 		azureSource.SecretName = csiSource.NodeStageSecretRef.Name
 		azureSource.SecretNamespace = &csiSource.NodeStageSecretRef.Namespace
-		if csiSource.VolumeAttributes != nil {
-			if shareName, ok := csiSource.VolumeAttributes[azureFileShareName]; ok {
-				azureSource.ShareName = shareName
-			}
-		}
-	} else {
+	}
+	if azureSource.ShareName == "" || azureSource.SecretName == "" {
 		rg, storageAccount, fileShareName, _, err := getFileShareInfo(csiSource.VolumeHandle)
 		if err != nil {
 			return nil, err
 		}
-		azureSource.ShareName = fileShareName
-		azureSource.SecretName = fmt.Sprintf(secretNameTemplate, storageAccount)
+		if azureSource.ShareName == "" {
+			azureSource.ShareName = fileShareName
+		}
+		if azureSource.SecretName == "" {
+			azureSource.SecretName = fmt.Sprintf(secretNameTemplate, storageAccount)
+		}
 		resourceGroup = rg
 	}
 
+	if azureSource.SecretNamespace == nil {
+		ns := defaultSecretNamespace
```

Comment:
this may need to be adjusted as part of undoing the regression in https://github.com/kubernetes/kubernetes/pull/97417

see https://github.com/kubernetes/kubernetes/issues/99061#issuecomment-801953931 for details

Reply:
in csi migration translation, there is no way to get pod namespace, unless we address this, otherwise it CSI migration would break also. We are planning to turn on csi migration in 1.21, it will break in that scenario too.

The main issue is that original inline azure file volume does not have a `secretNamespace` field, while azure file persistent volume has that field.

Reply:
the backwards compatibility issue has to be solved before that can be enabled

Reply:
is it possible to add a new field(`secretNamespace`) in azure file inline volume, that's the best way I could see to solve this issue. In csi migration translation lib, there is no way to get pod namespace.

Reply:
> is it possible to add a new field(`secretNamespace`) in azure file inline volume, that's the best way I could see to solve this issue.

No, secret references in pod specs cannot reference cross-namespace secrets.

> In csi migration translation lib, there is no way to get pod namespace.

That sounds like an issue that could be solved by plumbing the pod namespace through the go API.

Reply:
@andyzhangx do you need the secret only for node operations or also attach? If only node, then plumbing in pod namespace should not be difficult:
https://github.com/kubernetes/kubernetes/blob/54449be03127f883e52df48b95adef36925ae2cf/pkg/kubelet/volumemanager/populator/desired_state_of_world_populator.go#L556

Plumbing it through attach will be more effort as we would need to add a field to the VolumeAttachment:
https://github.com/kubernetes/kubernetes/blob/54449be03127f883e52df48b95adef36925ae2cf/pkg/apis/storage/types.go#L164

cc @Jiawei0227

Reply:
hi @msau42 we need pod namespace in mount(node operation), expand & delete(controller operations), is that possible? thanks.

Reply:
Expand and delete operations are not supported for inline volumes.